qep-314-coding-style.md

```diff
@@ -78,7 +78,19 @@ Notes:
 - 3.1. For readability and ease of code review, avoid use of ``auto``. The following exceptions are permitted:
 
   - 3.1.1. ``auto`` should be used for complex types, such as iterators. Eg ``for ( auto it = object.begin(); ...)``
-  
+  - 3.1.2. ``auto`` may be used for variable types if the type is explicit during variable initialization. Eg
+
+```
+// allowed, as the QgsPoint type is explicit during initialization:
+auto pointObject = QgsPoint( 3, 4 );
+// allowed, as the std::unique_ptr< QgsPoint >, std::shared_ptr< QgsPoint > types are explicit during initialization:
+auto pointUniquePointer = std::make_unique< QgsPoint >( 3, 4 );
+auto pointSharedPointer = std::make_shared< QgsPoint >( 3, 4 );
+
+// NOT allowed, the argument types for the std::tuple are not explicit:
+auto myTuple = std::make_tuple( 0, 5 );
```

Comment:
I think the best practice in this kind of document is to provide, when possible, the recommended/mandatory alternative when something is disallowed. So maybe the following? (Not totally sure if striking will work)

```suggestion
// NOT allowed, the argument types for the std::tuple are not explicit:
<s>auto myTuple = std::make_tuple( 0, 5 );</s>
// rather do
auto myTuple = std::tuple<int, int>( 0 , 5 )
```